ipfx/qc_features.py

```diff
@@ -371,23 +371,33 @@ def current_clamp_sweep_qc_features(sweep_data,sweep_info,ontology):
 
     return sweep
 
-def sweep_completed(i,v,hz):
+
+def sweep_completed(i,v,t,hz):
 
     POSTSTIM_STABILITY_EPOCH = 0.5
     LONG_RESPONSE_DURATION = 5  # this will count long ramps as completed
 
-    stimulus_end_ix = np.nonzero(i)[0][-1]  # last non-zero index along the only dimension=0
-    response_end_ix = np.nonzero(v)[0][-1]  # last non-zero index along the only dimension=0
+    di = np.diff(i)
+    di_idx = np.flatnonzero(di)   # != 0
+
+    if len(di_idx) < 4: # assuming there is a test pulse there should be at least 2 pulses (i.e, 4 jumps)
+        return False
+
+    stim_start_time, stim_duration, stim_amplitude, stim_start_idx, stim_end_idx = stf.get_stim_characteristics(i,t)
+    expt_start_ix, expt_end_ix = stf.get_experiment_epoch(i, v, hz)
+
+    stimulus_end_ix = stim_end_idx
+    response_end_ix = expt_end_ix
 
     post_stim_response_duration = (response_end_ix - stimulus_end_ix) / hz
     completed_expt_epoch = post_stim_response_duration > POSTSTIM_STABILITY_EPOCH
 
     long_response = response_end_ix/hz>LONG_RESPONSE_DURATION
```

Comment:
Change to `long_response = (response_end_ix / hz) > LONG_RESPONSE_DURATION` for better readability.

Also, although hz is likely a float, I feel like `from __future__ import division` should be considered since all of this is running on py2.7 right?